internal/services/applicationinsights/application_insights_resource.go

```diff
@@ -414,6 +408,141 @@ func resourceApplicationInsightsRead(d *pluginsdk.ResourceData, meta interface{}
 	return nil
 }
 
+func resourceApplicationInsightsUpdate(d *pluginsdk.ResourceData, meta interface{}) error {
+	client := meta.(*clients.Client).AppInsights.ComponentsClient
+	billingClient := meta.(*clients.Client).AppInsights.BillingClient
+	ctx, cancel := timeouts.ForUpdate(meta.(*clients.Client).StopContext, d)
+	defer cancel()
+
+	id, err := components.ParseComponentID(d.Id())
+	if err != nil {
+		return err
+	}
+
+	existing, err := client.ComponentsGet(ctx, *id)
+	if err != nil {
+		if !response.WasNotFound(existing.HttpResponse) {
+			return fmt.Errorf("checking for presence of existing %s: %v", id, err)
+		}
+	}
+	if existing.Model == nil {
+		return fmt.Errorf("retrieving %s: `model` was nil", id)
+	}
+	if existing.Model.Properties == nil {
+		return fmt.Errorf("retrieving %s: `properties` was nil", id)
+	}
+
+	componentProps := existing.Model.Properties
+
+	oldWorkspaceId, newWorkspaceId := d.GetChange("workspace_id")
+	if oldWorkspaceId.(string) != "" && newWorkspaceId.(string) == "" {
+		return fmt.Errorf("`workspace_id` cannot be removed after set. If `workspace_id` is not specified but you encounter a diff, this might indicate a Microsoft initiated automatic migration from classic resources to workspace-based resources. If this is the case, please update `workspace_id` in your config file to the new value.")
+	}
+
+	if d.HasChange("sampling_percentage") {
+		componentProps.SamplingPercentage = pointer.To(d.Get("sampling_percentage").(float64))
+	}
+
+	if d.HasChange("disable_ip_masking") {
+		componentProps.DisableIPMasking = pointer.To(d.Get("disable_ip_masking").(bool))
+	}
+
+	if d.HasChange("local_authentication_disabled") {
+		componentProps.DisableLocalAuth = pointer.To(d.Get("local_authentication_disabled").(bool))
+	}
+
+	if d.HasChange("internet_ingestion_enabled") {
+		if d.Get("internet_ingestion_enabled").(bool) {
+			componentProps.PublicNetworkAccessForIngestion = pointer.To(components.PublicNetworkAccessTypeEnabled)
+		} else {
+			componentProps.PublicNetworkAccessForIngestion = pointer.To(components.PublicNetworkAccessTypeDisabled)
+		}
```

Comment:
To stay consistent with other resources, I think here we should set `componentProps.PublicNetworkAccessForIngestion` to the default then overwrite if the condition is true

```suggestion
		componentProps.PublicNetworkAccessForIngestion = pointer.To(components.PublicNetworkAccessTypeDisabled)
		if d.Get("internet_ingestion_enabled").(bool) {
			componentProps.PublicNetworkAccessForIngestion = pointer.To(components.PublicNetworkAccessTypeEnabled)
		}
```